controller/share_manager_controller.go

```diff
@@ -438,10 +435,10 @@ func (c *ShareManagerController) syncShareManagerVolume(sm *longhorn.ShareManage
 		// ensure volume gets detached if share manager needs to be cleaned up and hasn't stopped yet
 		// we need the isNotNeeded var so we don't accidentally detach manually attached volumes,
 		// while the share manager is no longer running (only run cleanup once)
-		if isNotNeeded && sm.Status.State != types.ShareManagerStateStopped {
+		if isNotNeeded && sm.Status.State != types.ShareManagerStateStopping && sm.Status.State != types.ShareManagerStateStopped {
 			getLoggerForShareManager(c.logger, sm).Info("stopping share manager")
 			if err = c.detachShareManagerVolume(sm); err == nil {
-				sm.Status.State = types.ShareManagerStateStopped
+				sm.Status.State = types.ShareManagerStateStopping
 			}
```

Comment:
NIT: print out the error log

Reply:
The error log is already handled in the detachShareManagerVolume.
The only other error that is possible is conflict, which we don't need to print.